File: app/src/main/java/com/android/ichooseyou/model/UserManager.java
package com.android.ichooseyou.model;

import java.util.ArrayList;
import java.util.List;
import java.util.function.Predicate;

public enum UserManager {
    INSTANCE;

    private List<User> userList;

    UserManager() {
        userList = new ArrayList<>();
    }

    public void updateUser(User oldUser, User newUser) {
        int index = userList.indexOf(oldUser);
        if (index != -1) {
            userList.set(index, newUser);
        }
    }

    public boolean isUserDuplicate(String username) {
        return userList.stream().anyMatch(user -> user.getName().equals(username));
    }

    public boolean isUserExists(String username) {
        return userList.stream().anyMatch(user -> user.getName().equals(username));
    }

    public boolean isValidLogin(String username, String password) {
        return userList.stream().anyMatch(user -> user.getName().equals(username) && user.getPassword().equals(password));
    }

    public List<User> getUserList() {
        return userList;
    }
}

File: app/src/main/java/com/android/ichooseyou/model/UserManager.kt
package com.android.ichooseyou.model

enum class UserManager {
    INSTANCE;

    private val userList: MutableList<User> = mutableListOf()

    fun updateUser(oldUser: User, newUser: User) {
        val index = userList.indexOf(oldUser)
        if (index != -1) {
            userList[index] = newUser
        }
    }

    fun add(user: User) {
        userList.add(user)
    }

    fun isUserDuplicate(username: String): Boolean {
        return userList.any { user -> user.name == username }
    }

    fun isUserExists(username: String): Boolean {
        return userList.any { user -> user.name == username }
    }

    fun isValidLogin(username: String, password: String): Boolean {
        return userList.any { user -> user.name == username && user.password == password }
    }

    fun getUserList(): MutableList<User> {
        return userList
    }
}